test/k8s_tests/knative_e2e_test.go

```diff
@@ -0,0 +1,278 @@
+package k8s_tests
+
+import (
+	"fmt"
+	rcsv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	mock "github.com/dana-team/container-app-operator/test/k8s_tests/mocks"
+	utilst "github.com/dana-team/container-app-operator/test/k8s_tests/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	knativeMetricAnnotation = "autoscaling.knative.dev/metric"
+	imageExample            = "danateam/autoscale-go"
+	nonExistingImageExample = "example-python-app:v1"
+	exampleAppName          = "new-app-name"
+	newSecretKey            = "username"
+	creatorAnnotation       = "serving.knative.dev/creator"
+	lastModifierAnnotation  = "serving.knative.dev/lastModifier"
+	cappServiceAccount      = "system:serviceaccount:capp-operator-system:capp-operator-controller-manager"
+	exampleDanaAnnotation   = "rcs.dana.io/app-name"
+	testContainerName       = "capp-test-container"
+)
+
+// updateCapp updates the given Capp object and ensures the readiness of the latest revision
+// if shouldRevisionBeReady is true. It also checks and asserts the state of the LatestReadyRevision.
+func updateCapp(capp *rcsv1alpha1.Capp, shouldRevisionBeReady bool) {
+	latestReadyRevisionBeforeUpdate := capp.Status.KnativeObjectStatus.ConfigurationStatusFields.LatestReadyRevisionName
+	nextRevisionName := utilst.GetNextRevisionName(latestReadyRevisionBeforeUpdate)
```

Comment:
shouldn't it be only in the shouldbeready ?

Reply:
No, I still have to check that the revision is being created even if it shouldn't be ready